overlord/ifacestate/helpers.go

```diff
@@ -251,8 +252,10 @@ func (m *InterfaceManager) removeSnapSecurity(task *state.Task, snapName string)
 }
 
 type connState struct {
-	Auto      bool   `json:"auto,omitempty"`
-	Interface string `json:"interface,omitempty"`
+	Auto             bool                   `json:"auto,omitempty"`
+	Interface        string                 `json:"interface,omitempty"`
+	DynamicPlugAttrs map[string]interface{} `json:"dynamic-plug-attrs,omitempty"`
+	DynamicSlotAttrs map[string]interface{} `json:"dynamic-slot-attrs,omitempty"`
```

Comment:
These might be "plug-dynamic" and "slot-dynamic" as well, reusing the terms from the task.